src/main/java/org/broadinstitute/hellbender/tools/spark/transforms/bqsr/BaseRecalibratorEngineSpark.java

```diff
@@ -0,0 +1,125 @@
+package org.broadinstitute.hellbender.tools.spark.transforms.bqsr;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.engine.ReadContextData;
+import org.broadinstitute.hellbender.engine.ReferenceDataSource;
+import org.broadinstitute.hellbender.engine.ReferenceMemorySource;
+import org.broadinstitute.hellbender.engine.ContextShard;
+import org.broadinstitute.hellbender.engine.spark.AuthHolder;
+import org.broadinstitute.hellbender.utils.gcs.BucketUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
+import org.broadinstitute.hellbender.utils.recalibration.BaseRecalibrationEngine;
+import org.broadinstitute.hellbender.utils.recalibration.QuantizationInfo;
+import org.broadinstitute.hellbender.utils.recalibration.RecalUtils;
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationArgumentCollection;
+import org.broadinstitute.hellbender.utils.recalibration.RecalibrationTables;
+import org.broadinstitute.hellbender.utils.recalibration.covariates.StandardCovariateList;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+
+/**
+ * A lightweight wrapper over BaseRecalibrationEngine to make it easier to use from Spark.
+ * Takes in reads + contextual data (overlapping reference bases and variants), spits out RecalibrationTables.
+ */
+public final class BaseRecalibratorEngineSpark implements Serializable {
```

Comment:
Name `BaseRecalibrationEngineSparkWrapper` for clarity (ie., to avoid giving the impression that this is a copy of the engine itself)